Sources/LinkedIn/Core/HttpClient.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  HttpClient.swift
+//  PovioKitAuth
+//
+//  Created by Borut Tomazin on 22/05/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+struct HttpClient {
+  func request(method: String, url: URL, headers: [Header]) async throws -> Data {
+    var urlRequest = URLRequest(url: url)
```

Comment:
I would move serialization and deserialization into this function.

Reply:
I don't see any benefit by doing that.

Reply:
https://github.com/poviolabs/PovioKitAuth/pull/30/commits/a3d953122be4fa969af23a13bc2b8621922112d8